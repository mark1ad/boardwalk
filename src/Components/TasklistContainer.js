import React from "react";
import { PropTypes } from "prop-types";

const TasklistContainer = (props) => {
  return (
    <div className="tasklist-container">
      <div className="tasklist-container-name">
        {props.tasklist.name}
      </div>
    </div>
  );
};

TasklistContainer.propTypes = {
  tasklist: PropTypes.object.isRequired
};

export default TasklistContainer;
