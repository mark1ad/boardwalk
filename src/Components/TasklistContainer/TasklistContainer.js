import React from "react";
import { PropTypes } from "prop-types";
import ShowTaskCard from "../ShowTaskCard";
import ShowCreateTaskCard from "../ShowCreateTaskCard";

/**
 * Displays the tasks in a tasklist.
 */
const TasklistContainer = (props) => {
  return (
    <div className="tasklist-container">
      <div className="tasklist-container-name">
        {props.tasklist.name}
      </div>
      { props.tasklist.tasks.map(task =>
        (
          <ShowTaskCard key={task.id} task={task} />
        )
      )}
      <ShowCreateTaskCard tasklist_id={props.tasklist.id} />
    </div>
  );
};

TasklistContainer.propTypes = {
  /** List of tasks */
  tasklist: PropTypes.object.isRequired
};

export default TasklistContainer;
