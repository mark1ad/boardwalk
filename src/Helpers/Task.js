class Task {
  constructor(task) {
    this.id = task.id;
    this.name = task.name;
  }
}

export default Task;
