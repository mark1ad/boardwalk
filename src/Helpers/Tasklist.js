import Task from "./Task";

class Tasklist {
  constructor(tasklist) {
    this.id = tasklist.tasklist.id;
    this.name = tasklist.tasklist.name;
    this.tasks = [];
    tasklist.tasks.forEach(dbTask => {
      const task = new Task(dbTask);
      this.tasks.push(task);
    });
  }
}

export default Tasklist;
