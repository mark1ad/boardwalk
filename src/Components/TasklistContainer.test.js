import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Tasklist from '../Helpers/Tasklist';
import Task from '../Helpers/Task';
import TasklistContainer from './TasklistContainer';

let tasklist;
const task1 = new Task({id: 1, name: "task1"});
const task2 = new Task({id: 2, name: "task2"});
const tasks = [ task1, task2 ];

beforeEach(() => {
  tasklist = new Tasklist({id: 1, name: "name"});
})

it('renders without crashing', () => {
  shallow(<TasklistContainer tasklist={tasklist} />);
})

it("renders correctly with no tasks in tasklist", () => {
  const tree = shallow(<TasklistContainer tasklist={tasklist} />);
  expect(tree).toMatchSnapshot();
})

it("renders correctly with 2 tasks in tasklist", () => {
  tasklist.tasks = tasks;
  const tree = shallow(<TasklistContainer tasklist={tasklist} />);
  expect(tree).toMatchSnapshot();
})
