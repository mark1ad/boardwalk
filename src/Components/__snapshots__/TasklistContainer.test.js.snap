// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with 2 tasks in tasklist 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TasklistContainer
    tasklist={
        Tasklist {
            "id": 1,
            "name": "name",
            "tasks": Array [
              Task {
                "id": 1,
                "name": "task1",
              },
              Task {
                "id": 2,
                "name": "task2",
              },
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="tasklist-container-name"
>
          name
</div>,
        Array [
          <Connect(Card)
            task={
                        Task {
                                    "id": 1,
                                    "name": "task1",
                                  }
            }
/>,
          <Connect(Card)
            task={
                        Task {
                                    "id": 2,
                                    "name": "task2",
                                  }
            }
/>,
        ],
      ],
      "className": "tasklist-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "name",
          "className": "tasklist-container-name",
        },
        "ref": null,
        "rendered": "name",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "task": Task {
            "id": 1,
            "name": "task1",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "task": Task {
            "id": 2,
            "name": "task2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="tasklist-container-name"
>
            name
</div>,
          Array [
            <Connect(Card)
              task={
                            Task {
                                          "id": 1,
                                          "name": "task1",
                                        }
              }
/>,
            <Connect(Card)
              task={
                            Task {
                                          "id": 2,
                                          "name": "task2",
                                        }
              }
/>,
          ],
        ],
        "className": "tasklist-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "name",
            "className": "tasklist-container-name",
          },
          "ref": null,
          "rendered": "name",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "task": Task {
              "id": 1,
              "name": "task1",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "task": Task {
              "id": 2,
              "name": "task2",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders correctly with no tasks in tasklist 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TasklistContainer
    tasklist={
        Tasklist {
            "id": 1,
            "name": "name",
            "tasks": Array [],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="tasklist-container-name"
>
          name
</div>,
        Array [],
      ],
      "className": "tasklist-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "name",
          "className": "tasklist-container-name",
        },
        "ref": null,
        "rendered": "name",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="tasklist-container-name"
>
            name
</div>,
          Array [],
        ],
        "className": "tasklist-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "name",
            "className": "tasklist-container-name",
          },
          "ref": null,
          "rendered": "name",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
